lib/ansible/modules/windows/win_audit_rule.ps1

```diff
@@ -0,0 +1,182 @@
+#!powershell
+
+# Copyright: (c) 2017, Noah Sparks <nsparks@outlook.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy.psm1
+#Requires -Module Ansible.ModuleUtils.SID.psm1
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+# these are your module parameters
+$path = Get-AnsibleParam -obj $params -name "path" -type "str" -failifempty $true -aliases "destination","dest"
```

Comment:
There is a `path` type that automatically expands environment variables which would be better to use here.

Reply:
Thanks for the in depth review. I should have better investigated the type options. I'll try and work through all of these items tomorrow and validate functionality with the type changes.

Reply:
No worries, I see a lot of similarities to win_acl which is an older module and doesn't really have any of these features. Let me know once you are ready for another review and I'll do some actual testing once you're done.